experimental/packages/opentelemetry-exporter-prometheus/src/PrometheusExporter.ts

```diff
@@ -87,8 +87,11 @@ export class PrometheusExporter extends MetricReader {
 
     if (config.preventServerStart !== true) {
       this.startServer()
-        .then(callback)
-        .catch(err => diag.error(err));
+        .then(() => callback?.())
```

Comment:
You might even be able to do this:

```suggestion
        .then(callback, callback)
```

might require changing the constructor to 

```typescript
constructor(config: ExporterConfig = {}, callback: (error?: Error) => void = () => {})
```

Reply:
Ah yes, that improves readability. :slightly_smiling_face: Had to change the type of the callback argument `Error | void` instead to make it work with the argument type that `then()` on a `Promise<void>` expects. 

I still kept the `diag.error()` though. 
Otherwise troubleshooting may be hard for anyone who did not add a callback and is wondering why telemetry is not being exported. :thinking: